src/fiatExchanges/CoinbasePaymentSection.tsx

```diff
@@ -1,29 +1,63 @@
 import React from 'react'
 
+import { generateOnRampURL } from '@coinbase/cbpay-js'
+import { useAsync } from 'react-async-hook'
 import { useTranslation } from 'react-i18next'
 import { Image, StyleSheet, Text, View } from 'react-native'
 import { useSelector } from 'react-redux'
 import { coinbasePayEnabledSelector } from 'src/app/selectors'
 import Touchable from 'src/components/Touchable'
 import { FetchProvidersOutput } from 'src/fiatExchanges/utils'
+import { readOnceFromFirebase } from 'src/firebase/firebase'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
 import colors from 'src/styles/colors'
 import fontStyles from 'src/styles/fonts'
+import { CiCoCurrency } from 'src/utils/currencies'
+import { walletAddressSelector } from 'src/web3/selectors'
 
 export interface CoinbasePaymentSectionProps {
+  digitalAsset: CiCoCurrency
+  cryptoAmount: number
   coinbaseProvider: FetchProvidersOutput | undefined
 }
 
-export function CoinbasePaymentSection({ coinbaseProvider }: CoinbasePaymentSectionProps) {
+export function CoinbasePaymentSection({
+  digitalAsset,
+  cryptoAmount,
+  coinbaseProvider,
+}: CoinbasePaymentSectionProps) {
   const { t } = useTranslation()
   const coinbasePayEnabled = useSelector(coinbasePayEnabledSelector)
+  const walletAddress = useSelector(walletAddressSelector)!
+  const allowedAssets = [CiCoCurrency.CELO]
 
-  if (!coinbaseProvider || coinbaseProvider.restricted || !coinbasePayEnabled) {
+  const appIdResponse = useAsync(async () => readOnceFromFirebase('coinbasePay/appId'), [])
+  const appId = appIdResponse.result
+
+  if (
+    !coinbaseProvider ||
+    coinbaseProvider.restricted ||
+    !coinbasePayEnabled ||
+    !allowedAssets.includes(digitalAsset)
```

Comment:
I think you should add `!appId` to this list. If for some reason we can't fetch appId from firebase, we probably don't want to display coinbase as an option

Reply:
addressed!